qiskit_metal/toolbox_metal/layer_stack_handler.py

```diff
@@ -0,0 +1,439 @@
+# -*- coding: utf-8 -*-
+
+from textwrap import fill
+from xmlrpc.client import Boolean
+import pandas as pd
+from typing import List, Tuple
+from typing import Union
+from addict import Dict
+import os
+import logging
+
+from .parsing import parse_units, TRUE_STR, FALSE_STR
+
+
+class LayerStackHandler():
+    """Use DataFrame to hold information for multiple chips.
+
+    For all the chips, do NOT reuse the layer-number.  However, within the dataframe,
+    we can repeat layer numbers since a layer can have multiple datatypes.
+
+    """
+
+    def __init__(self,
+                 multi_planar_design: 'MultiPlanar',
+                 fname: Union['str', None] = None) -> None:
+        """Use the information in MultiPlanar design to parse_value
+        and get the name of filename for layer stack.
+
+        Args:
+            multi_planar_design (MultiPlanar): Class that is child of QDesign.
+        """
+
+        self.multi_planar_design = multi_planar_design
+        self.logger = self.multi_planar_design.logger  # type: logging.Logger
+
+        self.filename_csv_df = None
+        if hasattr(self.multi_planar_design,
+                   'ls_filename') and self.multi_planar_design.ls_filename:
+            #populate pandas table from data in this file.
+            self.filename_csv_df = self.multi_planar_design.ls_filename
+        elif fname:
+            self.filename_csv_df = fname
+
+        self.ls_df = None
+        # self.column_names = [
+        #     'chip_name', 'layer', 'datatype', 'material', 'thickness',
+        #     'z_coord', 'fill'
+        # ]
+        self.layer_stack_default = Dict(chip_name=['main'],
+                                        layer=[1],
+                                        datatype=[0],
+                                        material=['pec'],
+                                        thickness=['2um'],
+                                        z_coord=['0um'],
+                                        fill=['false'])
```

Comment:
I'm not completely sure, but shouldn't this be "true" instead of "false"? Because the whole metal volume would be filled by "pec" right?

Reply:
Good point, will change it.  I had used both just for development.  Will change the default to true.

Reply:
Done -committed.